snapcraft/internal/errors.py

```diff
@@ -133,6 +133,24 @@ def __init__(self, *, base, linker_path):
         super().__init__(base=base, linker_path=linker_path)
 
 
+class IncompatibleBaseError(SnapcraftError):
+    fmt = (
+        "The linker version {linker_version!r} used by the base {base!r} is "
+        "incompatible with all or some of the files in this snap:\n"
```

Comment:
You're kindly giving a list, so I think this can be reduced to "[...] is incompatible with files in this snap: [...]